[INFO   ] Logger: Record log in /storage/emulated/0/Work/TheArchitect/Tools/.kivy/logs/kivy_21-11-02_42.txt
[INFO   ] Kivy: v2.0.0
[INFO   ] Kivy: Installed at "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/kivy/__init__.py"
[INFO   ] Python: v3.9.7 (default, Oct  6 2021, 01:34:26) 
[GCC 11.1.0]
[INFO   ] Python: Interpreter at "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/bin/python3"
[INFO   ] Factory: 186 symbols loaded
[INFO   ] Image: Providers: img_tex, img_dds, img_sdl2 (img_pil, img_ffpyplayer ignored)
[INFO   ] Text: Provider: sdl2
[INFO   ] Window: Provider: sdl2
[INFO   ] GL: Using the "OpenGL ES 2" graphics system
[INFO   ] GL: Backend used <sdl2>
[INFO   ] GL: OpenGL version <b'OpenGL ES 3.2 V@444.0 (GIT@0defdeb, I18bc8d115f, 1599640306) (Date:09/09/20)'>
[INFO   ] GL: OpenGL vendor <b'Qualcomm'>
[INFO   ] GL: OpenGL renderer <b'Adreno (TM) 619'>
[INFO   ] GL: OpenGL parsed version: 3, 2
[INFO   ] GL: Texture max size <16384>
[INFO   ] GL: Texture max units <16>
[INFO   ] Window: auto add sdl2 input provider
[INFO   ] Window: virtual keyboard not allowed, single mode, not docked
[INFO   ] GL: NPOT texture support is available
[WARNING] stderr: Traceback (most recent call last):
[WARNING] stderr:   File "/storage/emulated/0/Work/TheArchitect/Tools/devprojectmgr.py", line 45, in build
[WARNING] stderr:     self.workspaceRoot=e.getExternalStorageDirectory()
[WARNING] stderr:   File "kivy/properties.pyx", line 498, in kivy.properties.Property.__set__
[WARNING] stderr:   File "kivy/properties.pyx", line 542, in kivy.properties.Property.set
[WARNING] stderr:   File "kivy/properties.pyx", line 533, in kivy.properties.Property.set
[WARNING] stderr:   File "kivy/properties.pyx", line 699, in kivy.properties.StringProperty.check
[WARNING] stderr: ValueError: ProjectApp.workspaceRoot accept only str
[WARNING] stderr: 
[WARNING] stderr: During handling of the above exception, another exception occurred:
[WARNING] stderr: 
[WARNING] stderr: Traceback (most recent call last):
[WARNING] stderr:   File "/storage/emulated/0/Work/TheArchitect/Tools/devprojectmgr.py", line 70, in <module>
[WARNING] stderr:     ProjectApp().run()
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/kivy/app.py", line 949, in run
[WARNING] stderr:     self._run_prepare()
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages/kivy/app.py", line 919, in _run_prepare
[WARNING] stderr:     root = self.build()
[WARNING] stderr:   File "/storage/emulated/0/Work/TheArchitect/Tools/devprojectmgr.py", line 46, in build
[WARNING] stderr:     except a:
[WARNING] stderr: NameError: name 'a' is not defined
